Utils/Extensions/NSURLExtensions.swift

```diff
@@ -181,6 +181,31 @@ extension NSURL {
         }
     }
 
+    public func displayURL() -> NSURL? {
+        if self.isReaderModeURL() {
+            return self.decodeReaderModeURL()?.havingRemovedAuthorisationComponents()
+        }
+
+        if self.isErrorPageURL() {
+            let decodedURL = self.originalURLFromErrorPageURL()
+            if !decodedURL!.isAboutURL() {
+                return decodedURL?.havingRemovedAuthorisationComponents()
+            } else {
+                return nil
+            }
+        }
+
+        if !self.isAboutURL() {
+            return self.havingRemovedAuthorisationComponents()
+        }
+
+        return nil
+    }
+
+    public func displayString() -> String? {
```

Comment:
Same. Also, it might not be worth having this in the first place; we only use `url.displayString()` in one place. Maybe just remove this and use `url.displayURL?.absoluteString`?